ptrace.c

```diff
@@ -463,6 +471,8 @@ bool sm_searchregions(globals_t *vars, scan_match_type_t match_type, const userv
         show_info("use the \"reset\" command to refresh regions.\n");
         return sm_detach(vars->target);
     }
+
+    INTERRUPTABLESCAN();
```

Comment:
Shouldn't this go right before the `while (n) {`?
At the very least this block:
```
     vars->scan_progress = 0.0;
     vars->stop_flag = false;
     n = vars->regions->head;
```
has to go before it, or we risk leaving user-visible things in a mixed state.

Reply:
Good point, not sure why I put it that far up.